File: solutions/2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.swift
class Solution {
    func findGCD(_ nums: [Int]) -> Int {
        var mn = nums[0], mx = nums[0]
        for x in nums {
            if x < mn { mn = x }
            if x > mx { mx = x }
        }
        var a = mn, b = mx
        while b != 0 {
            let t = a % b
            a = b
            b = t
        }
        return a
    }
}

File: solutions/2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.java
class Solution {
    public int findGCD(int[] nums) {
        int mn = nums[0], mx = nums[0];
        for (int x : nums) {
            if (x < mn) mn = x;
            if (x > mx) mx = x;
        }
        int a = mn, b = mx;
        while (b != 0) {
            int t = a % b;
            a = b;
            b = t;
        }
        return a;
    }
}